bridge/src/reactor.rs

```diff
@@ -0,0 +1,124 @@
+use crate::connection::{Connection, ConnectionError};
+use tokio::runtime::Builder;
+use tokio::sync::{mpsc, oneshot};
+
+#[derive(Debug)]
+pub enum Request {
+    Connect {
+        host: String
+    },
+    Rpc {
+        connection: Connection,
+        method: String,
+        bytes: Vec<u8>
+    }
+}
+
+#[derive(Debug)]
+pub enum Response {
+    Empty {},
+    Connection {
+        connection: Connection
+    },
+    Rpc {
+        bytes: Vec<u8>
+    }
+}
+
+#[derive(Debug)]
+pub enum ReactorError {
```

Comment:
Same comment here about the impls / probably not needing to be a specifically defined enum since we don't match on it

Reply:
Using anyhow will also allow you to get rid of all these `map_err`s